### ISSUE TITLE
Multiple queries under a single root query not passing all data to props

### ISSUE THREAD

**Issue by Bouncey** [gatsbyjs/gatsby#1459]
This is my query in `Article.jsx` which is a template component 
```js
export const pageQuery = graphql`
query ArticleBySlug($slug: String!) {
  markdownRemark(fields: { slug: { eq: $slug }}) {
    html
    frontmatter {
      title
    }
  }
  # get all site pages to pass to parent to render nav
  allMarkdownRemark {
    edges {
      node {
        frontmatter {
          title
          parent
        }
      }
    }
  }
}
`;
```
I only get `markdownRemark` in `this.props.data` and not `allMarkdownRemark`

Am I missing something?

I there an easier way to get what I am after in my `Layout.jsx`?

**Comment by Bouncey** [comment]
To solve this, I have taken to creating a JSON object which I import into my layout component.

**Comment by KyleAMathews** [comment]
This should work. Does the query work in graphiql?

**Comment by KyleAMathews** [comment]
Wait layout? Graphql queries aren't yet supported in layout queries.

**Comment by Bouncey** [comment]
Yes the query works in graphiql. I found an issue relating to queries in layout, so I tried a dual query in my article template with a view of passing the data up to my layout component. I don't think that changing queries is triggering a cache bust, as seen in other scenarios.

I understand now that this would be too much of an overhead as it would attach the `allMarkdownRemark` data to every single page. Which with over 700 pages just isn't workable.
